### DISCUSSION TITLE
[@types/p5] p5.Image.pixels should be Uint8ClampedArray, not a simple number array.

### DISCUSSION THREAD

**Discussion by SanzSeraph** [DefinitelyTyped/DefinitelyTyped#57372] [Issues with a @types package]
Per the p5.js docs, Image.pixels is a Uint8ClampedArray. This is important, since Uint8ClampedArray lacks certain methods available on the plain JavaScript array class, and possesses certain methods that the JS array class does not have.

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment]
Could you please provide documentation?

**Reply by SanzSeraph** [reply]
https://p5js.org/reference/#/p5/pixels

**Reply by SanzSeraph** [reply]
Before you call loadPixels(), "pixels" appears to be a normal array. But if you call loadPixels() then inspect the pixels variable, it becomes a Uint8ClampedArray, as described in the documentation above.

**Reply by SanzSeraph** [reply]
I should also add that this is an important distinction, because since pixel is a Uint8ClampedArray, that means you can use its native set method to set all of its values in one go based on an iterable, array-like object. I ran across this because I was using WebWorkers to optimize some image generation logic, and in my brief experimentation, p5.Image objects cannot be passed around using Worker messaging. I didn't want to have to manually iterate through all of the elements of the source array in order to set the values of the pixels variable, because I suspected that would largely defeat the purpose of using WebWorkers to begin with.
